tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1292,3 +1292,145 @@ def test_max(is_module, xp, device, input, axis, dtype):
+        in_dtype, out_dtype = dtypes
+        try:
+            a_np = input.astype(in_dtype)
+        except (ValueError, OverflowError):
+            return xp.zeros(())  # invalid combination of data and dtype
+
+        a = xp.array(a_np)
+        if xp is numpy:
+            a = dtype_utils.cast_if_numpy_array(xp, a, out_dtype)
+        return xp.mean(a, axis)
+
+    @chainerx.testing.numpy_chainerx_array_equal(strides_check=False)
+    @pytest.mark.parametrize('input,axis', [
+        # --- single axis
+        # input, axis
+        # valid params
+        (numpy.asarray(0), None),
+        (numpy.asarray(-1), None),
+        (numpy.asarray(float('inf')), None),
+        (numpy.asarray(float('nan')), None),
+        (numpy.asarray(-float('inf')), None),
+        (numpy.asarray([4, 1, 4, 1]), None),
+        (numpy.asarray([4, 1, 4, 1]), 0),
+        (numpy.asarray([[4, 4, 1, 1], [4, 1, 4, 1]]), 0),
+        (numpy.asarray([[4, 4, 1, 1], [4, 1, 4, 1]]).T, 1),
+        (numpy.asarray([-0.0, +0.0, +0.0, -0.0]), None),
+        (numpy.asarray([[True, True, False, False],
+                        [True, False, True, False]]), 0),
+        (numpy.ones((2, 0, 3)), 2),
+        (numpy.ones((2, 3)), 1),
+        (numpy.ones((2, 3)), -2),
+        # --- multiple axes
+        # input, axis
+        # valid params
+        (numpy.asarray([[1, 4, 3, 1], [4, 6, 3, 2], [2, 3, 6, 1]]), (0, 1)),
+        (numpy.asarray([[1, 4, 3, 1], [4, 6, 3, 2], [2, 3, 6, 1]]), (-2, -1)),
+    ])
+    @pytest.mark.parametrize('dtypes', _expected_dtypes_math_functions)
+    @pytest.mark.parametrize_device(['native:0', 'cuda:0'])
+    # TODO: Remove strides_check=False
+    def test_valid_mean(self, xp, device, input, axis, dtypes):
+        return self.apply_func(xp, device, input, axis, dtypes)
+
+    @chainerx.testing.numpy_chainerx_array_equal(
+        accept_error=(IndexError, ValueError, chainerx.DimensionError),
+        strides_check=False)
+    @pytest.mark.parametrize('input,axis', [
+        # --- single axis
+        # input, axis
+        # invalid params
+        (numpy.ones((0,)), None),
+        (numpy.ones((2, 0, 3)), 1),
+        (numpy.ones((2, 0, 3)), None),
+        (numpy.ones((2, 3)), 2),
+        (numpy.ones((2, 3)), -3),
+        # --- multiple axes
+        # input, axis
+        # invalid params
+        (numpy.asarray([[1, 4, 3, 1], [4, 6, 3, 2], [2, 3, 6, 1]]), (1, 1)),
+        (numpy.asarray([[1, 4, 3, 1], [4, 6, 3, 2], [2, 3, 6, 1]]), (-3, 1)),
+        (numpy.asarray([[1, 4, 3, 1], [4, 6, 3, 2], [2, 3, 6, 1]]), (1, 2)),
+    ])
+    @pytest.mark.parametrize('dtypes', _expected_dtypes_math_functions)
+    @pytest.mark.parametrize_device(['native:0', 'cuda:0'])
+    # TODO: Remove strides_check=False
+    def test_invalid_mean(self, xp, device, input, axis, dtypes):
+        return self.apply_func(xp, device, input, axis, dtypes)
+
+
+class TestVar(object):
+    def apply_func(self, xp, device, input, axis, dtypes):
```

Comment:
Does this have to be a method? You could make it a free function and reuse it for `mean` and `var` test. In that case, you don't need these "Test" prefixed classes neither.

Reply:
Have refactored the test code for it to be a function. Please a look